### DISCUSSION TITLE
Open Source Vs Closed Source

### DISCUSSION THREAD

**Discussion by vcta99** [albertkun/21S-ASIAAM-191A#121] [Past Thinking Caps]
Generally, I believe that closed source software serves to add a layer of obscurity between the user and the software. Reducing this transparency can be dangerous for several reasons: improvements to the software take longer and are harder to come by (decreased pool of those who can access the code), decreased transparency leaves room for intentional exploitation of the user, etc. Open source software, while not perfect, are at least subject to review, and therefore encourage a higher level of critique and discussion. I also think that open source software stays truer to the original hopes and aspirations of the internet (they sure were optimists!), as a space for collaboration and democratic values. I think of open source software similarly to free speech. We all have free speech in this country, but that does not stop some people from generating harmful/misinformed rhetoric). And while clauses can be introduced to open source software that prohibits use in bad faith (similar to how hate speech is technically not tolerated), I worry about how effectively these clauses can be enforced.

**Comment by latishasumardy** [comment]
I agree with your points on closed source software. I think that the secrecy of it can sometimes serve to be more harmful than good, especially when the user does not really understand the software and what it does fully. This not only may lead to companies exploiting users, but also users may run into errors that they don't know how to fix but the company may not have enough developers to fix the problem. Open source software is really great because it allows for a large community of users that are invested in finding and fixing bugs they find in the code, and with the open source code they can create new inventive programs based on the original code.
